src/Emulator/Visuals/Visuals.js

```diff
@@ -1,11 +1,236 @@
+  const [elementBounds, setElementBounds] = useState({top: contentDimensions.height || 125, bottom: contentDimensions.height|| 125, left: '', right: contentDimensions.width || 125})
+
+  const [maxRes, setMaxRes] = useState({
+    height: null,
+    width: null,
+    prevHeight: null,
+    prevWidth: null});
+
+  function convert2Px(containerDimensions) {
+    const movableSpaceReg = /([\d]+)+/g
+    const movableSpaceStringH = containerDimensions.height.toString()
+    const movableSpaceStringW = containerDimensions.width.toString()
+    const movableSpaceH = movableSpaceStringH.match(movableSpaceReg)[0]
+    const movableSpaceW = movableSpaceStringW.match(movableSpaceReg)[0]
+    const movableSpacePxH = window.innerHeight * movableSpaceH / 100
+    const movableSpacePxW = window.innerWidth * movableSpaceW / 100    
+    return {movableSpacePxH, movableSpacePxW}
+  }
+
+  function convert2Vh(movableSpace) {
+    const hundred = 100
+    const movableSpaceH = movableSpace / window.innerHeight * hundred + 'vh'
+    setContainerDimensions({
+      height: movableSpaceH, width: containerWidth
+    }) 
+    return movableSpaceH
+  }
+
+  useEffect(() => {
+    const padding = 20
+    const movableSpace = convert2Px(containerDimensions)
+    const elementBot = parseInt(handleDetect.y) + parseInt(elementBounds.bottom) + padding
+    console.log(elementBounds.bottom)
+    
+    if (elementBot > movableSpace.movableSpacePxH) {
+      const addSpace = 50
+      const addMovableSpace = movableSpace.movableSpacePxH + addSpace
+      const newMovableSpace = convert2Vh(addMovableSpace)
+    }
+    console.log(elementBot, movableSpace.movableSpacePxH, containerDimensions.height)
+  }, [handleDetect, containerDimensions, elementBounds, contentDimensions])
+
+  // Handles dragging
+  const handleDrag = (e, ui) => {
+    const {x, y} = pos;
+    setPos({
+        x: x + ui.deltaX,
+        y: y + ui.deltaY,
+    });
+
+    var dragging = document.getElementsByClassName('handle')[0].style.transform;
+    var draggedPos = /([\d]+)+/g
+    var dragX = dragging.match(draggedPos)
+    
+    setHandleDetect({
+      x: dragX[0],
+      y: dragX[1]
+    })
+  };
+  
+  // Handles what to do when starting drag and stopping drag
+  const onStart = () => {
+    setDrags(drags => drags + 1);
+  };
+
+  const onStop = () => {
+    setDrags(drags => drags - 1);
+  };
+
+  // When image is added, adjusts the container and drag handler sizes to accomodate for image size
+  const imageLoad = () => {
+    setContentDimensions({
+      height: imageRef.current.offsetHeight + additionalImageSpace,
+      width: imageRef.current.offsetWidth + additionalImageSpace,
+    });
+    setElementBounds({
+      top: imageRef.current.offsetHeightt + additionalImageSpace,
+      bottom: imageRef.current.offsetWidth + additionalImageSpace,
+      left: '',
+      right: imageRef.current.offsetWidth + additionalImageSpace
+    })
```

Comment:
make this a helper function where you can specify whether to zoom out (add) or zoom in (subtract) by `additionalImageSpace` (good job making it a constant, perhaps call it `zoomIncrementSize` or something)